src/overlays/actors/ovl_En_Bba_01/z_en_bba_01.c

```diff
@@ -33,84 +34,296 @@ const ActorInit En_Bba_01_InitVars = {
+    /* Sword beam     */ DMG_ENTRY(0, BBA01_DMG_EFF_NONE),
+    /* Normal Roll    */ DMG_ENTRY(0, BBA01_DMG_EFF_NONE),
+    /* UNK_DMG_0x1B   */ DMG_ENTRY(0, BBA01_DMG_EFF_NONE),
+    /* UNK_DMG_0x1C   */ DMG_ENTRY(0, BBA01_DMG_EFF_NONE),
+    /* Unblockable    */ DMG_ENTRY(0, BBA01_DMG_EFF_NONE),
+    /* UNK_DMG_0x1E   */ DMG_ENTRY(0, BBA01_DMG_EFF_NONE),
+    /* Powder Keg     */ DMG_ENTRY(0, BBA01_DMG_EFF_NONE),
 };
 
-#endif
+u16 D_809CCCD8[] = { 4000, 4, 1, 3, 6000, 4, 1, 6, 4000, 4, 1, 3, 6000, 4, 1, 6 };
+
+void EnBba01_UpdateModel(EnBba01* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    Vec3f focus;
+
+    EnHy_UpdateSkelAnime(&this->enHy, globalCtx);
+    if (func_8013D5E8(this->enHy.actor.shape.rot.y, 0x36B0, this->enHy.actor.yawTowardsPlayer)) {
+        focus.x = player->actor.world.pos.x;
+        focus.y = player->bodyPartsPos[7].y + 3.0f;
+        focus.z = player->actor.world.pos.z;
+        func_8013D2E0(&focus, &this->enHy.actor.focus.pos, &this->enHy.actor.shape.rot, &this->enHy.focusTarget,
+                      &this->enHy.headRot, &this->enHy.torsoRot, D_809CCCD8);
+    } else {
+        Math_SmoothStepToS(&this->enHy.focusTarget.x, 0, 4, 0x3E8, 1);
+        Math_SmoothStepToS(&this->enHy.focusTarget.y, 0, 4, 0x3E8, 1);
+        Math_SmoothStepToS(&this->enHy.headRot.x, 0, 4, 0x3E8, 1);
+        Math_SmoothStepToS(&this->enHy.headRot.y, 0, 4, 0x3E8, 1);
+        Math_SmoothStepToS(&this->enHy.torsoRot.x, 0, 4, 0x3E8, 1);
+        Math_SmoothStepToS(&this->enHy.torsoRot.y, 0, 4, 0x3E8, 1);
+    }
+    SubS_FillLimbRotTables(globalCtx, this->enHy.limbRotTableY, this->enHy.limbRotTableZ,
+                           ARRAY_COUNT(this->enHy.limbRotTableY));
+    EnHy_UpdateCollider(&this->enHy, globalCtx);
+}
+
+s32 EnBba01_TestIsTalking(EnBba01* this, GlobalContext* globalCtx) {
+    s32 isTalking = false;
+
+    if (Actor_ProcessTalkRequest(&this->enHy.actor, &globalCtx->state)) {
+        isTalking = true;
+        this->enHy.textId = 0x10B9;
+        this->enHy.tmpFocusTarget = this->enHy.focusTarget;
+        this->enHy.tmpHeadRot = this->enHy.headRot;
+        this->enHy.tmpTorsoRot = this->enHy.torsoRot;
+        this->enHy.tmpActionFunc = this->enHy.actionFunc;
+        this->enHy.actionFunc = EnBba01_Talk;
+    }
+    return isTalking;
+}
+
+s32 func_809CC270(EnBba01* this, GlobalContext* globalCtx) {
+    s16 x;
+    s16 y;
+
+    Actor_GetScreenPos(globalCtx, &this->enHy.actor, &x, &y);
+    //! @bug: Both x and y conditionals are always true, || should be an &&
+    if (!this->enHy.waitingOnInit && ((x >= 0) || (x < SCREEN_WIDTH)) && ((y >= 0) || (y < SCREEN_HEIGHT))) {
+        func_800B85E0(&this->enHy.actor, globalCtx, 30.0f, EXCH_ITEM_2E);
+    }
+    return true;
+}
+
+void EnBba01_FinishInit(EnHy* this, GlobalContext* globalCtx) {
+    //! @bug: gBbaSkel does not match EnHy's skeleton assumptions.
+    //! Main offender is that gBbaSkel has 18 limbs, while EnHy expects 16
```

Comment:
Are there more details worth putting here?

Reply:
I could also mention that since the expected limbs is smaller, the joint and morph tables are too small so will overflow. 

The other kind of buggy aspect is that seen below in the Draw functions, the limbs don't match with what you expect. Thus for example, instead of her head and torso following Link, it is instead her right arm and bag that follow him. Though this is probably more because the whole actor seems to have been written with OOT's object_bba in mind, and this never got updated when the model was.

Reply:
Yeah, all of those sound like appropriate to add in various places - I'd even make a general description under the comment block since this is so mis-implemented: sounds like it requires significant modification to use properly.

Reply:
Adjusted the bug comment to now mention the overflow, also added a note to the header about the other buggy limb behavior.